### ISSUE TITLE
v2.0 support: RowanBrowserService >> recompileMethodsAfterClassCompilation will need to change

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#825]
changing RowanBrowserService >> recompileMethodsAfterClassCompilation may depend upon the ultimate resolution of GemTalk/Rowan#626. My current plans involve allowing unmanaged classes and methods to coexist with package classes and methods ... the implication here is that the Project browser should not expect that all classes and methods will be packaged. 

As I'm starting to make this a reality, I've hit this error in the RowanServices code:
```
```
```
a UserDefinedError occurred (error 2318), reason:halt, No loaded package named '(NONE)' found

[1] UserDefinedError (AbstractException) >> _signalWith: @5 line 25
[2] UserDefinedError (AbstractException) >> signal @2 line 47
[3] RwGsImage class (Object) >> error: @6 line 7
[4] [] in RwGsImage class >> loadedPackageNamed: @9 line 5
[5] RwGsImage class >> loadedPackageNamed:ifAbsent: @4 line 11
[6] RwGsImage class >> loadedPackageNamed: @2 line 4
[7] RwPackage >> _loadedPackage @4 line 3
[8] RwPackage >> isDirty @2 line 5
[9] RowanPackageService >> rowanDirty @3 line 3
[10] RowanPackageService >> isDirty @2 line 3
[11] RowanPackageService class >> forPackageNamed: @7 line 7
[12] RowanBrowserService >> recompileMethodsAfterClassCompilation @11 line 15
[13] RowanBrowserService (Object) >> perform:withArguments: @1 line 12
[14] RowanBrowserService >> servicePerform:withArguments: @2 line 2
[15] [] in JadeServer >> updateFromSton: @35 line 13
[16] OrderedCollection (Collection) >> do: @5 line 10
[17] [] in JadeServer >> updateFromSton: @24 line 9
[18] ExecBlock0 (ExecBlock) >> on:do: @3 line 44
[19] [] in JadeServer >> updateFromSton: @12 line 14
[20] ExecBlock0 (ExecBlock) >> on:do: @3 line 44
[21] JadeServer64bit35 (JadeServer) >> updateFromSton: @2 line 23
[22] GsNMethod class >> _gsReturnToC @1 line 11
===============================================
[1] UserDefinedError (AbstractException) >> _signalWith: @5 line 25
	receiver oop(152045825) == a UserDefinedError occurred (error 2318), reason:halt, No loaded package named '(Ne MNUONE)' found

instance variables:
	-currGsHandler oop(20) == nil
	-gsArgs oop(20) == nil
	-gsDetails oop(152066305) == 'No loaded package named ''(NONE)'' found'
	-gsNumber oop(18546) == 2318
	-gsReason oop(152066561) == 'halt'
	-gsResumable oop(268) == true
	-gsStack oop(20) == nil
	-gsTrappable oop(268) == true
	-messageText oop(152059905) == 'a UserDefinedError occurred (error 2318), reason:halt, No loaded package named ''(NONE)'' found'
	-tag oop(20) == nil

temporary variables and parameters:
	inCextensionArg oop(20) == nil
	res oop(20) == nil
----------------------------------------------------
[2] UserDefinedError (AbstractException) >> signal @2 line 47
	receiver oop(152045825) == a UserDefinedError occurred (error 2318), reason:halt, No loaded package named '(NONE)' found

instance variables:
	-currGsHandler oop(20) == nil
	-gsArgs oop(20) == nil
	-gsDetails oop(152066305) == 'No loaded package named ''(NONE)'' found'
	-gsNumber oop(18546) == 2318
	-gsReason oop(152066561) == 'halt'
	-gsResumable oop(268) == true
	-gsStack oop(20) == nil
	-gsTrappable oop(268) == true
	-messageText oop(152059905) == 'a UserDefinedError occurred (error 2318), reason:halt, No loaded package named ''(NONE)'' found'
	-tag oop(20) == nil

temporary variables and parameters:
----------------------------------------------------
[3] RwGsImage class (Object) >> error: @6 line 7
	receiver oop(26900993) == RwGsImage

instance variables:

temporary variables and parameters:
	messageString oop(152066305) == 'No loaded package named ''(NONE)'' found'
----------------------------------------------------
[4] [] in RwGsImage class >> loadedPackageNamed: @9 line 5
	receiver oop(151177729) == anExecBlock0

instance variables:

temporary variables and parameters:
	aString oop(145089281) == '(NONE)'
	self oop(26900993) == RwGsImage
----------------------------------------------------
[5] RwGsImage class >> loadedPackageNamed:ifAbsent: @4 line 11
	receiver oop(26900993) == RwGsImage

instance variables:

temporary variables and parameters:
	absentBlock oop(151177729) == anExecBlock0
	aName oop(145089281) == '(NONE)'
----------------------------------------------------
[6] RwGsImage class >> loadedPackageNamed: @2 line 4
	receiver oop(26900993) == RwGsImage

instance variables:

temporary variables and parameters:
	aString oop(145089281) == '(NONE)'
----------------------------------------------------
[7] RwPackage >> _loadedPackage @4 line 3
	receiver oop(151171585) == aRwPackage for (NONE)

instance variables:
	-name oop(145089281) == '(NONE)'

temporary variables and parameters:
----------------------------------------------------
[8] RwPackage >> isDirty @2 line 5
	receiver oop(151171585) == aRwPackage for (NONE)

instance variables:
	-name oop(145089281) == '(NONE)'

temporary variables and parameters:
----------------------------------------------------
[9] RowanPackageService >> rowanDirty @3 line 3
	receiver oop(151176961) == aRowanPackageService:(NONE)

instance variables:
	-classes oop(20) == nil
	-command oop(20) == nil
	-commandArgs oop(20) == nil
	-defaultTemplate oop(20) == nil
	-definition oop(20) == nil
	-hierarchyServices oop(20) == nil
	-isDirty oop(20) == nil
	-name oop(145089281) == '(NONE)'
	-organizer oop(20) == nil
	-packageName oop(20) == nil
	-projectDefinition oop(20) == nil
	-projectName oop(20) == nil
	-selectedClass oop(20) == nil
	-testClasses oop(20) == nil
	-updates oop(20) == nil
	-updateType oop(20) == nil

temporary variables and parameters:
----------------------------------------------------
[10] RowanPackageService >> isDirty @2 line 3
	receiver oop(151176961) == aRowanPackageService:(NONE)

instance variables:
	-classes oop(20) == nil
	-command oop(20) == nil
	-commandArgs oop(20) == nil
	-defaultTemplate oop(20) == nil
	-definition oop(20) == nil
	-hierarchyServices oop(20) == nil
	-isDirty oop(20) == nil
	-name oop(145089281) == '(NONE)'
	-organizer oop(20) == nil
	-packageName oop(20) == nil
	-projectDefinition oop(20) == nil
	-projectName oop(20) == nil
	-selectedClass oop(20) == nil
	-testClasses oop(20) == nil
	-updates oop(20) == nil
	-updateType oop(20) == nil

temporary variables and parameters:
----------------------------------------------------
[11] RowanPackageService class >> forPackageNamed: @7 line 7
	receiver oop(26958337) == RowanPackageService

instance variables:

temporary variables and parameters:
	aName oop(145089281) == '(NONE)'
	inst oop(151176961) == aRowanPackageService:(NONE)
----------------------------------------------------
[12] RowanBrowserService >> recompileMethodsAfterClassCompilation @11 line 15
	receiver oop(151161857) == aRowanBrowserService

instance variables:
	-allClasses oop(151165185) == anOrderedCollection( )
	-command oop(28165889) == #'recompileMethodsAfterClassCompilation'
	-commandArgs oop(151164673) == anArray( )
	-definition oop(20) == nil
	-dictionaries oop(20) == nil
	-hierarchyServices oop(20) == nil
	-newCachedClasses oop(151165953) == anArray( )
	-newCachedSelectors oop(151165697) == anArray( )
	-organizer oop(151164929) == aClassOrganizer
	-projects oop(20) == nil
	-removedMethods oop(20) == nil
	-selectedClass oop(151165441) == 'XXX'
	-testCount oop(20) == nil
	-testPackages oop(20) == nil
	-updates oop(20) == nil
	-updateType oop(20) == nil

temporary variables and parameters:
	classService oop(151166209) == aRowanClassService:XXX
	packageService oop(20) == nil
	projectService oop(20) == nil
	theClass oop(152046849) == XXX
----------------------------------------------------
[13] RowanBrowserService (Object) >> perform:withArguments: @1 line 12
	receiver oop(151161857) == aRowanBrowserService

instance variables:
	-allClasses oop(151165185) == anOrderedCollection( )
	-command oop(28165889) == #'recompileMethodsAfterClassCompilation'
	-commandArgs oop(151164673) == anArray( )
	-definition oop(20) == nil
	-dictionaries oop(20) == nil
	-hierarchyServices oop(20) == nil
	-newCachedClasses oop(151165953) == anArray( )
	-newCachedSelectors oop(151165697) == anArray( )
	-organizer oop(151164929) == aClassOrganizer
	-projects oop(20) == nil
	-removedMethods oop(20) == nil
	-selectedClass oop(151165441) == 'XXX'
	-testCount oop(20) == nil
	-testPackages oop(20) == nil
	-updates oop(20) == nil
	-updateType oop(20) == nil

temporary variables and parameters:
	anArray oop(151164673) == anArray( )
	aSelectorSymbol oop(28165889) == #'recompileMethodsAfterClassCompilation'
----------------------------------------------------
[14] RowanBrowserService >> servicePerform:withArguments: @2 line 2
	receiver oop(151161857) == aRowanBrowserService

instance variables:
	-allClasses oop(151165185) == anOrderedCollection( )
	-command oop(28165889) == #'recompileMethodsAfterClassCompilation'
	-commandArgs oop(151164673) == anArray( )
	-definition oop(20) == nil
	-dictionaries oop(20) == nil
	-hierarchyServices oop(20) == nil
	-newCachedClasses oop(151165953) == anArray( )
	-newCachedSelectors oop(151165697) == anArray( )
	-organizer oop(151164929) == aClassOrganizer
	-projects oop(20) == nil
	-removedMethods oop(20) == nil
	-selectedClass oop(151165441) == 'XXX'
	-testCount oop(20) == nil
	-testPackages oop(20) == nil
	-updates oop(20) == nil
	-updateType oop(20) == nil

temporary variables and parameters:
	collection oop(151164673) == anArray( )
	symbol oop(28165889) == #'recompileMethodsAfterClassCompilation'
----------------------------------------------------
[15] [] in JadeServer >> updateFromSton: @35 line 13
	receiver oop(152058625) == anExecBlock1

instance variables:

temporary variables and parameters:
	organizer oop(151164929) == aClassOrganizer
	service oop(151161857) == aRowanBrowserService
----------------------------------------------------
[16] OrderedCollection (Collection) >> do: @5 line 10
	receiver oop(152059137) == anOrderedCollection( aRowanLoggingService, aRowanBrowserService)

instance variables:

temporary variables and parameters:
	aBlock oop(152058625) == anExecBlock1
	i oop(18) == 2
----------------------------------------------------
[17] [] in JadeServer >> updateFromSton: @24 line 9
	receiver oop(151163905) == anExecBlock0

instance variables:

temporary variables and parameters:
	services oop(152059137) == anOrderedCollection( aRowanLoggingService, aRowanBrowserService)
----------------------------------------------------
[18] ExecBlock0 (ExecBlock) >> on:do: @3 line 44
	receiver oop(151163905) == anExecBlock0

instance variables:

temporary variables and parameters:
	exceptionSelector oop(27095809) == GsInteractionRequest
	fastPath oop(268) == true
	handlerBlock oop(151162113) == anExecBlock
----------------------------------------------------
[19] [] in JadeServer >> updateFromSton: @12 line 14
	receiver oop(151181569) == anExecBlock0

instance variables:

temporary variables and parameters:
	organizer oop(151164929) == aClassOrganizer
	resultString oop(20) == nil
	self oop(152494081) == aJadeServer64bit35
	services oop(152059137) == anOrderedCollection( aRowanLoggingService, aRowanBrowserService)
	stonString oop(152047873) == 'OrderedCollection[RowanLoggingService{#command:#logReceivedServices,#commandArgs:[],#id:0,#groupId:309,#services:OrderedCollection[RowanBrowserService{#command:#recompileMethodsAfterClassCompilation,...
----------------------------------------------------
[20] ExecBlock0 (ExecBlock) >> on:do: @3 line 44
	receiver oop(151181569) == anExecBlock0

instance variables:

temporary variables and parameters:
	exceptionSelector oop(130817) == Error
	fastPath oop(268) == true
	handlerBlock oop(61605121) == anExecBlock1
----------------------------------------------------
[21] JadeServer64bit35 (JadeServer) >> updateFromSton: @2 line 23
	receiver oop(152494081) == aJadeServer64bit35

instance variables:
	-classList oop(20) == nil
	-classOrganizers oop(152451585) == anArray( nil, nil, nil, nil)
	-environment oop(2) == 0
	-methodCommandResult oop(20) == nil
	-methodFilters oop(20) == nil
	-methodFilterType oop(20) == nil
	-readStream oop(20) == nil
	-selectedClass oop(20) == nil
	-selections oop(20) == nil
	-writeStream oop(20) == nil

temporary variables and parameters:
	organizer oop(151164929) == aClassOrganizer
	resultString oop(20) == nil
	services oop(152059137) == anOrderedCollection( aRowanLoggingService, aRowanBrowserService)
	stonString oop(152047873) == 'OrderedCollection[RowanLoggingService{#command:#logReceivedServices,#commandArgs:[],#id:0,#groupId:309,#services:OrderedCollection[RowanBrowserService{#command:#recompileMethodsAfterClassCompilation,...
----------------------------------------------------
[22] GsNMethod class >> _gsReturnToC @1 line 11
	receiver oop(144897) == GsNMethod

instance variables:

temporary variables and parameters:
```
When executing the following class creation method:
![image](https://user-images.githubusercontent.com/623951/85323807-8eec8b80-b47d-11ea-8829-788cf3c60842.png)

Basically the changes that I'm making in Rowan are to tolerate unmanaged classes and `(NON)` is the indication that the class or method is unmanaged ...

**Comment by dalehenrich** [comment]
This is not a critical bug ... the classes and methods (presumably) are created correctly and I'm working through the errors in Rowan as we speak:)
